spring-data-cassandra/src/main/java/org/springframework/data/cassandra/core/mapping/CassandraPersistentProperty.java

```diff
@@ -179,4 +179,13 @@ default boolean isEmbedded() {
 	@Nullable
 	AnnotatedType findAnnotatedType(Class<? extends Annotation> annotationType);
 
+
+	/**
+	 * Whether the property is a static column.
+	 * @since 3.2
+	 */
+	default boolean isStaticColumn() {
```

Comment:
Annotation lookup can be somewhat expensive. It makes sense to implement the lookup directly in `BasicCassandraPersistentProperty` and update `CachingCassandraPersistentProperty` to cache the outcome whether that's a static column.

Reply:
Good catch! I've made the corresponding fix and updated `BasicCassandraPersistentTupleProperty` with `false` since it cannot be static. Additionally, I've added a separate test for a static tuple field.